### PR TITLE
[cmake] fix building with internal fmt

### PR DESCRIPTION
**Pull request by wsnipex**
fixes a build issue with internal fmt, see https://forum.kodi.tv/showthread.php?tid=333879

## Description

make the kodi target depend on fmt

## How Has This Been Tested?

built with and without internal libfmt on ubuntu

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [ ] None of the above (please explain below)

### CODE REVIEW

**Review by Rechi — APPROVED** [review]

**Comment by wsnipex** [comment]
anyone knows why uwp broke?

**Comment by Rechi** [comment]
found it
https://github.com/xbmc/xbmc/blob/740e800c2b3fe2b785450815f9d5bee84eeabf11/CMakeLists.txt#L139-L142

File: CMakeLists.txt [hunk 1]
```diff
@@ -139,6 +140,7 @@ endif()
 if(CORE_SYSTEM_NAME STREQUAL windowsstore)
   #uses header only version of Fmt
   list(REMOVE_ITEM required_deps Fmt)
+  list(REMOVE_ITEM CORE_REQUIRED_DEPS fmt)
```
Comment (Rechi) [inline CMakeLists.txt]: no this is the wrong fix, correct is removing this hole hack

Comment (wsnipex) [reply CMakeLists.txt]: I agree, but I have no idea why uwp uses this and I'm certainly not going to touch windows deps.

Comment (Rechi) [reply CMakeLists.txt]: then wait with merging, I'll PR the removal

Comment (wsnipex) [reply CMakeLists.txt]: great, thanks

**Review by lindspems85 — APPROVED** [review]

**Pull request merged**